SIMElasticity.h

```diff
@@ -387,7 +388,17 @@ template<class Dim> class SIMElasticity : public Dim
     for (; child; child = child->NextSiblingElement())
       if (this->parseDimSpecific(child))
         continue;
-
+      else if (!strcasecmp(child->Value(),"texturematerial"))
+      {
+        IFEM::cout <<"  Parsing <"<< child->Value() <<">"<< std::endl;
+        int code = this->parseMaterialSet(child,mVec.size());
+        IFEM::cout <<"\tMaterial code "<< code <<":";
+        bool planeStrain = Dim::dimension == 2 ? Elastic::planeStrain : true;
+        IsotropicTextureMat* mat = new IsotropicTextureMat(planeStrain,
+                                                           Elastic::axiSymmetry);
+        mat->parse(child);
+        mVec.push_back(mat);
+      }
```

Comment:
Consider merge this block with the "isotropic" equivalent below, i.e., instead of l402
```
  else if (!strcasecmp(child->Value(),"isotropic") ||
           !strcasecmp(child->Value(),"texturematerial"))
```
and then
```
  Material* mat;
  if (!strcasecmp(child->Value(),"texturematerial")) {
    mat = new IsotropicTextureMat(planeStrain,Elastic::axiSymmetry);
    mat->parse(child);
  }
  else
    mat = this->getIntegrand()->parseMatProp(child,planeStrain);
  mVec.push_back(mat);
```
Too much code duplication as of now. An even better solution would maybe to move the above inside `Elasticity::parseMatprop`.

Reply:
I need to get the restrictions removed first. There is a plethora of extensions to this tag coming up so I won't combine these two blocks until it is finalized. In no particular order I want to

* Don't assume `ASMs2D` patches; but I need to get the `ASMbase*` corresponding to a given topologyset and I need to get umin/umax from each of these patches. 
* Don't assume u=[0,1], but rather pass umin/umax to the texturematieral object itself so it can do appropriate coordinate transformations.
* Create `RealFunc`- or `VectorFunc` - functions from image intensity values to material properties
* Extend to volume case with 3D-tensor of image values (rgba)
* Move this from `IFEM-Elasticity` to some more general parsing (`IFEM-Common`? `IFEM`?)

OK, I take it back. I *want* to do in a particular order. The order given above...